source/agora/network/NetworkManager.d

```diff
@@ -162,11 +162,14 @@ public class NetworkManager
                     key = client.getPublicKey();
                     break;
                 }
-                catch (HTTPStatusException ex)
+                catch (Exception ex)
                 {
-                    // 404 => API not implemented, it's a FullNode
-                    if (ex.status == 404)
-                        break;
+                    if (auto http = cast(HTTPStatusException)ex)
+                    {
+                        // 404 => API not implemented, it's a FullNode
+                        if (http.status == 404)
+                            break;
+                    }
```

Comment:
How does it work with LocalRest ?

Reply:
It's the NetworkClient which throws the exception if there was a timeout.

Without this change and with the test-case I've added ordinarily this would happen:

```
#### FATAL ERROR: Request failure to Validator #2 (GDC2...YD36) after 2 attempts
This node was started at source/agora/test/Base.d:459
This most likely means that the node crashed due to an uncaught exception
If not, please file a bug at https://github.com/Geod24/localrest/
```